### PR TITLE
Fixes #8801: Inconsistent hashing algorithms used in passwords generated in the script engine

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/8801

### CODE REVIEW

**Comment by amousset** [comment]
Commit modified

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/JavascriptEngine.scala [hunk 1]
```diff
@@ -172,7 +172,7 @@ trait JsLibPassword extends ImplicitGetBytes {
   def aix(algo: String, s: String, salt: String): String = {
     algo.toLowerCase match {
       case "md5" => aixMd5(s)
```
Comment (peckpeck) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/JavascriptEngine.scala]: md5 with salt

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/JavascriptEngine.scala [hunk 2]
```diff
@@ -152,7 +152,7 @@ trait JsLibPassword extends ImplicitGetBytes {
   def unix(algo: String, s: String, salt: String): String = {
     algo.toLowerCase match {
       case "md5" => cryptMd5(s)
```
Comment (peckpeck) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/JavascriptEngine.scala]: md5 with salt

**Comment by jooooooon** [comment]
This looks good to me, thanks @amousset. I'd really like to get an opinion from @fanf or @VinceMacBuche before merging this though.

**Comment by VinceMacBuche** [comment]
Great @amousset !

**Comment by jooooooon** [comment]
Thanks @VinceMacBuche ! Enjoy the rest of your holiday :)

**Pull request merged**